### ISSUE TITLE
Quest Titles should not include "The" to accomodate A-Z sort and translations

### ISSUE THREAD

**Issue by lemoness** [HabitRPG/habitica#9361]
~In the Quests Shop, sorting from A-Z includes "The" as a criterion. Sorting should be tweaked to ignore "the" when it's the first word of a Quest name.~

Edited: The simplest solution for this issue would be to remove "the" from all quest titles. This preserves the functionality of the A-Z sort as well as our ability to allow for translations.

**Comment by Alys** [comment]
This is interesting from a translation perspective. For example, French
speakers would want Le and La ignored when sorting.  Coding that for every
language would be nightmarish.

I am wondering if the best solution would be to have a second title that's
used for sorting and then translators could handle this however best for
their language. E.g. in English, the second title would be "Birds of
Preycrastination".

I'm sure I've seen something similar done in some other app or website but
exactly where escapes me.

**Comment by clay-reed-a** [comment]
I do not believe it is that nightmarish. We create a hash map of languages to a list of leading words that language considers ignorable. If we know where our users are we know what languages to focus on. Similarly, doing it with a data structure like this makes adding a new “ignorable word” trivial. 

I do not think the secondary “sortable title” field is a good solution at all. It basically negates the whole point of having a computer sort the titles: the users sort the titles for the computers instead.

Clay a

**Comment by Alys** [comment]
I think users sorting the titles isn't bad because it's once-off effort for
each title and gives a good result for any leading word that needs to be
ignored, with no need for a code change to add leading words when new ones
are identified, but if you'd like to do the hash map solution, I have no
objection!

Something else that might be relevant: Is sorting currently being done
correctly for the RTL languages? If not, we should fix it but we could
create a new issue for that. Trimming should be handled correctly in this
issue though, if any of the RTL languages use leading words.

@clay-reed-a Do you want to work on a fix for this issue?

**Comment by TheHollidayInn** [comment]
I'm posting an initial fix for the. I believe it is the correct approach.

I think the general solution is similar to the NLP problem of removing stop words?

**Comment by TheHollidayInn** [comment]
Should I remove all stopwords (only from sort) depending on language: https://www.npmjs.com/package/stopword. Or do we want to just start our own list of definite articles.

**Comment by Alys** [comment]
I like the idea of using a stopwords package, especially one that handles multiple languages as that one does. When a player has their language set to non-English, we'd use only that language's stopwords for sorting.

Stopwords are also likely to be useful for searching for guilds and challenges: https://github.com/HabitRPG/habitica/issues/9755
When a player has their language set to non-English, we should probably use their language's stopwords AND English stopwords for searching (since they could be searching for either language-specific guilds/challenges or guilds/challenges written in English).

**Comment by clay-reed-a** [comment]
Seems I missed the party on this one. I would like to work on this website
tho. @Alys Could you assign me something? I do a lot of front end JS
interactivity stuff, and CSS perfecting.

**Comment by Alys** [comment]
@clay-reed-a Sorry about the delay! Have a look at issues tagged with our [help wanted](https://github.com/HabitRPG/habitica/labels/help%20wanted) label. You might find something there that suits you!

**Comment by mirrorcoloured** [comment]
Looks like this one has been waiting for a while. I've added the stopwords module which seems to provide a nice sort order:
![image](https://user-images.githubusercontent.com/2054119/196851891-a0278e8e-c148-49b8-80a4-8bc608c1ef19.png)
My working branch is here:
https://github.com/HabitRPG/habitica/compare/develop...mirrorcoloured:habitica:fix-az-sort
I have two questions:

1. I am only able to get one look at the site. After refreshing I get:
`...\habitica\website\client\src\components\shops\quests\index.vue
  406:1  error  'stopword' should be listed in the project's dependencies. Run 'npm i -S stopword' to add it  import/no-extraneous-dependencies`
I installed the module with `npm i stopword` which as far as I can tell automatically includes the -S flag now. I also tried with `-S` explicitly, and the behavior is the same. If I stop and restart the client I get no error the first time I refresh, but it returns after that. Is there any guidance on the proper way to add new modules to satisfy eslint?

2. To incorporate other languages, It looks like there will need to be a map between languages from Habitica → stopword. I see the list of languages in Habitica, but is there something like a `currentUser.language` to feed into a mapping?

**Comment by CuriousMagpie** [comment]
@SabreCat, can you answer @mirrorcoloured's questions?
